src/main/java/techreborn/blockentity/generator/basic/WindMillBlockEntity.java

```diff
@@ -42,13 +46,13 @@
 	public float bladeAngle;
 	public float spinSpeed;
 
-	public WindMillBlockEntity() {
-		super(TRBlockEntities.WIND_MILL);
+	public WindMillBlockEntity(BlockPos pos, BlockState state) {
+		super(TRBlockEntities.WIND_MILL, pos, state);
 	}
 
 	@Override
-	public void tick() {
-		super.tick();
+	public void tick(World world, BlockPos pos, BlockState state, MachineBaseBlockEntity blockEntity) {
```

Comment:
Are these tick methods not supposed to be static? Its been a while since I last looked at it so id have to check the code. Vanilla has a client and server one if im not mistaken, and your supposed to modify the data on the BE passed in as a param.

Reply:
The way vanilla does it is completely without an interface. Blocks have a `getTicker(...)` function now and that returns the static tick function.
I don't know about a client and a server one though.
I went for this solution (implementing BlockEntityTicker on the BlockEntity), so we can have one generic `getTicker(...)` in BaseBlockEntityProvider that calls the respective (non-static) `tick()` function of the BlockEntity.